Remotion/Web/ClientScript/Scripts/ListMenu.ts

```diff
@@ -0,0 +1,315 @@
+  public get RequiredSelection(): number { return this._requiredSelection; }
+  public get IsDisabled(): boolean { return this._isDisabled; }
+  public get Href(): string { return this._href; }
+  public get Target(): string { return this._target; }
+  public get DiagnosticMetadata(): Dictionary<string | boolean> { return this._diagnosticMetadata; }
+  public get DiagnosticMetadataForCommand(): Dictionary<string | boolean> { return this._diagnosticMetadataForCommand; }
+
+  constructor(
+    id: string,
+    category: string,
+    text: string,
+    icon: string,
+    iconDisabled: string,
+    requiredSelection: number,
+    isDisabled: boolean,
+    href: string,
+    target: string,
+    diagnosticMetadata: Dictionary<string | boolean>,
+    diagnosticMetadataForCommand: Dictionary<string | boolean>)
+  {
+    this._id = id;
+    this._category = category;
+    this._text = text;
+    this._icon = icon;
+    this._iconDisabled = iconDisabled;
+    this._requiredSelection = requiredSelection;
+    this._isDisabled = isDisabled;
+    this._href = href;
+    this._target = target;
+    this._diagnosticMetadata = diagnosticMetadata;
+    this._diagnosticMetadataForCommand = diagnosticMetadataForCommand;
+  }
+}
+
+class ListMenu
+{
+  private readonly _itemClassName: string = 'listMenuItem';
+  private readonly _itemFocusClassName: string = 'listMenuItemFocus';
+  private readonly _itemDisabledClassName: string = 'listMenuItemDisabled';
+  private readonly _requiredSelectionAny: number = 0;
+  private readonly _requiredSelectionExactlyOne: number = 1;
+  private readonly _requiredSelectionOneOrMore: number = 2;
+  private _listMenuInfos: Dictionary<ListMenuMenuInfo> = {};
+
+  public Initialize ($listMenu: JQuery): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsJQuery("$listMenu", $listMenu);
+
+    $listMenu.keydown ((event) =>
+    {
+      this.OnKeyDown (event, $listMenu);
+    });
+  }
+
+  public AddMenuInfo (listMenu: Element, menuInfo: ListMenuMenuInfo)
+  {
+    this._listMenuInfos[listMenu.id] = menuInfo;
+  }
+
+  public Update (listMenu: Element, getSelectionCount: () => number)
+  {
+    var menuInfo = this._listMenuInfos[listMenu.id];
+    if (menuInfo == null)
+      return;
+
+    var itemInfos = menuInfo.ItemInfos;
+    var selectionCount = -1;
+    if (getSelectionCount != null)
+      selectionCount = getSelectionCount();
+
+    for (var i = 0; i < itemInfos.length; i++)
+    {
+      var itemInfo = itemInfos[i];
+      var isEnabled = true;
+      if (itemInfo.IsDisabled)
+      {
+        isEnabled = false;
+      }
+      else
+      {
+        if (itemInfo.RequiredSelection == this._requiredSelectionExactlyOne && selectionCount != 1)
+        {
+          isEnabled = false;
+        }
+        if (itemInfo.RequiredSelection == this._requiredSelectionOneOrMore && selectionCount < 1)
+        {
+          isEnabled = false;
+        }
+      }
+      var item = <Element>document.getElementById (itemInfo.ID);
```

Comment:
`item` should have a null check - bug?

Reply:
No, this never be null so long as no one manipulates the DOM. These cases, I would simply make a Damit operator and be done with it in C#. @a-ctor how do we do this in TS?

Reply:
Same way as in C# with a bang!